.github/workflows/build.yml

```
@@ -0,0 +1,40 @@
+name: Gradle Build Project Trigger
+
+on:
+  pull_request:
+    branches: [dev, main]
+    paths-ignore:
+      - '**.gradle/wrapper'
+      - '**.gitignore'
+      - 'LICENSE'
+      - 'THIRD PARTY NOTICES'
+      - '**.md'
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps: 
+    - uses: actions/checkout@v3
+    - name: Setup JDK
+      uses: actions/setup-java@v3
+      with: 
+        java-version: 17
+        distribution: 'adopt'
+        cache: gradle
+    - name: Detect Secrets
+      uses: RobertFischer/detect-secrets-action@v2.0.0
+    - name: Build with Gradle
+      run: ./gradlew --no-daemon build
+    - name: Upload Unit Test Results
+      uses: actions/upload-artifact@v3
```

```suggestion
      if: ${{ always() }}
      uses: actions/upload-artifact@v3
```

we want to upload the test results, especially when they fail

---

omitting the `if` means always AFAIK

---

omitting it means only if the previous step was successful, which might not be the case when unit tests fail for instance.

---

you are right, thanks for pointing out 👍